ambry-frontend/src/integration-test/java/com/github/ambry/frontend/S3IntegrationTest.java

```diff
@@ -0,0 +1,282 @@
+/**
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.frontend;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.xml.XmlMapper;
+import com.github.ambry.account.Account;
+import com.github.ambry.account.Container;
+import com.github.ambry.account.InMemAccountService;
+import com.github.ambry.account.InMemAccountServiceFactory;
+import com.github.ambry.clustermap.MockClusterMap;
+import com.github.ambry.commons.CommonTestUtils;
+import com.github.ambry.commons.LoggingNotificationSystem;
+import com.github.ambry.commons.NettySslFactory;
+import com.github.ambry.commons.SSLFactory;
+import com.github.ambry.commons.TestSSLUtils;
+import com.github.ambry.config.FrontendConfig;
+import com.github.ambry.config.MySqlNamedBlobDbConfig;
+import com.github.ambry.config.NettyConfig;
+import com.github.ambry.config.SSLConfig;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.frontend.s3.S3MessagePayload;
+import com.github.ambry.rest.NettyClient;
+import com.github.ambry.rest.RestServer;
+import com.github.ambry.utils.TestUtils;
+import io.netty.handler.codec.http.DefaultHttpHeaders;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.HttpHeaders;
+import io.netty.handler.codec.http.HttpMethod;
+import io.netty.handler.codec.http.HttpResponse;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.netty.handler.codec.http.HttpUtil;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.security.GeneralSecurityException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Properties;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import static com.github.ambry.frontend.s3.S3MessagePayload.*;
+import static com.github.ambry.rest.RestUtils.Headers.*;
+import static com.github.ambry.rest.RestUtils.*;
+import static org.junit.Assert.*;
+
+
+/**
+ * Integration tests for Ambry S3 support.
+ */
+public class S3IntegrationTest extends FrontendIntegrationTestBase {
```

Comment:
Thanks for adding the integration test.
Do you plan to add more test including the List, Get, Head, Delete etc?

Reply:
Yup, will add more in coming days/weeks.